service/pager/query_pagers.cc

```diff
@@ -141,16 +142,19 @@ future<result<service::storage_proxy::coordinator_query_result>> query_pager::do
         // last ck can be empty depending on whether we
         // deserialized state or not. This case means "last page ended on
         // something-not-bound-by-clustering" (i.e. a static row, alone)
-        const bool has_ck = _has_clustering_keys && _last_ckey;
+        const bool has_ck = _has_clustering_keys && _last_pos.region() == partition_region::clustered;
 
         // If we have no clustering keys, it should mean we only have one row
         // per PK. Thus we can just bypass the last one.
         modify_ranges(_ranges, lo, has_ck, dht::ring_position_comparator(*_schema));
 
         if (has_ck) {
             query::clustering_row_ranges row_ranges = _cmd->slice.default_row_ranges();
-            clustering_key_prefix ckp = clustering_key_prefix::from_exploded(*_schema, _last_ckey->explode(*_schema));
-            query::trim_clustering_row_ranges_to(*_schema, row_ranges, ckp, reversed);
+            position_in_partition_view next_pos = _last_pos;
+            if (_last_pos.has_key()) {
+                next_pos = reversed ? position_in_partition_view::before_key(_last_pos) : position_in_partition_view::after_key(_last_pos);
+            }
+            query::trim_clustering_row_ranges_to(*_schema, row_ranges, next_pos, reversed);
```

Comment:
You adjust the bound, but only if has_key(). Shouldn't  an adjustment be done regardless of the partition region? For example, if we were in the static row region, we should advance to the clustering region (before all rows).

Reply:
The static row is always included in the query result, even if we read just the last row in the partition.
This adjusted position is only used to adjust the clustering row ranges in the slice, positions outside of the clustering domain has no impact on this.